Use iComfort client code from module

danielmcq opened the pull request:
I'm happy to see a real world use of my iComfort client in action. When I realized that it had been used in this project, I updated the client code to get it ready for publishing in npm as a module. This pull request is for an update to the alexa-icomfort project to have it use the [icomfort](https://github.com/deHugo/icomfort-js) module as a dependency.

danielmcq committed:
Use iComfort package from npm

danielmcq committed:
Fix iComfort client instantiation

danielmcq committed:
Update reference to iComfort repo

kate-hall commented:
@danielmcq ahh!! This is great, thank you so much (and thanks for developing the client in the first place). I'll test this branch on my thermostat tonight before merging.

kate-hall commented:
@danielmcq Annoyingly, AWS Lambda doesn't yet support all of the ES6 functionality, particularly some you're using in the module (http://whatdoeslambdasupport.com/). Right now I'm getting this error:
<img width="422" alt="screen shot 2017-02-14 at 3 58 10 pm" src="https://cloud.githubusercontent.com/assets/8494775/22954983/841db28e-f2ce-11e6-9a4f-ed2bbe59d5bf.png">

This one's related to the lack of support for default parameters.

 **EDIT** Ignore! Changing just that one line lets it run again! I'll make the change in my local module.

danielmcq commented:
Good to know about AWS Lambda. I wasn't aware of the limitation.

Shall we close this pull request?

kate-hall commented:
Nope, let's leave it open until I can find a way to make it work, I like the idea of moving to your module, just investigating options!

Commit:
Update icomfort dependency version

danielmcq commented:
@kate-hall I've updated the icomfort module to version 1.1.1. I made sure to test it with Node 4.3.2, which is the version which [AWS Lambda uses](http://docs.aws.amazon.com/lambda/latest/dg/current-supported-versions.html). I updated this pull request to reflect that change. I believe this will correctly work for you now.

kate-hall commented:
Awesome! I'll test this evening.

kate-hall commented on src/index.js:
```diff
@@ -132,7 +133,7 @@ function handleControl(event, context) {
 
             // send the change request to Lennox, send a response to Alexa on promise fulfillment
-            iComfort.setThermostatInfo(getThermostatInfoListParams, newParams.toSet, auth)
+            iComfort.setThermostatInfo(getThermostatInfoListParams, newParams.toSet)
```
@danielmcq I believe this line should be `iComfort.setThermostatInfo(newParams.toSet)` for compatibility with the node module, if you want to adjust?

kate-hall commented on tests/index.js:
```diff
@@ -30,21 +32,21 @@ var getSystemsInfoParams = {UserId: auth.username};
 
 /* Test 4 = Careful! This test will change your actual temperature, uncomment to run */
-// iComfort.setThermostatInfo(getInfoParams, changeTemperatureTestVars, auth)
+// iComfort.setThermostatInfo(getInfoParams, changeTemperatureTestVars)
```
@danielmcq Ditto the above, this line should be `iComfort.setThermostatInfo(changeTemperatureTestVars)` for compatibility with the node module, if you want to adjust?
